tracing-core/src/collect.rs

```diff
@@ -461,11 +462,7 @@ impl dyn Collect {
     pub fn downcast_ref<T: Any>(&self) -> Option<&T> {
         unsafe {
             let raw = self.downcast_raw(TypeId::of::<T>())?;
-            if raw.is_null() {
-                None
-            } else {
-                Some(&*(raw as *const _))
-            }
+            Some(&*(raw.as_ptr() as *const _))
```

Comment:
style, take it or leave it: this is equivalent to
```suggestion
            Some(&*(raw.cast::<T>().as_ptr()))
```
it's up to you if you think that looks nicer or not. :)